sdk/data/azcosmos/cosmos_client.go

```diff
@@ -48,7 +56,16 @@ func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (
 	if err != nil {
 		return nil, err
 	}
-	return &Client{endpoint: endpoint, pipeline: newPipeline(newCosmosBearerTokenPolicy(cred, scope, nil), o)}, nil
+
+	internalClient := &Client{endpoint: endpoint, pipeline: newInternalPipeline(newCosmosBearerTokenPolicy(cred, scope, nil), o), gem: &globalEndpointManager{}}
```

Comment:
you don't need 2 clients, just 2 pipelines. You can have both pipelines in the same `Client` and then when you can pass the internal pipeline to `newGlobalEndpointManager` + the endpoint. Inside Global Endpoint Manager, instead of using the client to send the request, you can send it through the internal pipeline (which contains the auth too).

Reply:
I was using an internal client as opposed to an internal pipeline here since regardless the call made within the Update() method are to the client.sendGetRequest() method, which also calls client.createRequest() to create the request we will send and client.executeAndEnsureSuccessResponse() to send said request - and only then in this method do we use the pipeline.

I figured the refactoring of all of this into whatever logic we would need to only use the internal pipeline would end up being overkill (separate logic for creating/sending the request), when in the end the way the request would be sent is through a pipeline anyway. I figured since it would be the same in terms of number of connections, doing it this way would be more straightforward. Maybe I'm missing something here though, happy to get a proper explanation if that's the case - not sure if there's some additional overhead I'm not seeing here.

Reply:
I don't know why I thought the methods were a lot bigger - went ahead and pushed the updates, thanks for pointing this out 👍